### PR TITLE
Pic 2568

### PR DESCRIPTION
**Pull request by rschallapalli**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingEntity.java [hunk 1]
```diff
@@ -59,7 +47,6 @@ public class HearingEntity extends BaseImmutableEntity implements Serializable {
     private final Long id;
 
     @Column(name = "HEARING_ID", nullable = false)
-    @Setter // TODO: This was added to enable ImmutableCourtCaseService.enforceValidHearingId() as a precautionary measure and should be removed ASAP
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingEntity.java:62]: Good riddance 🙄

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepository.java [hunk 2]
```diff
@@ -53,36 +53,24 @@ public interface HearingRepository extends CrudRepository<HearingEntity, Long>{
...
-        "       inner join hearing_day on hearing_day.fk_hearing_id = group_hearing.id " +
-        "       where group_hearing.created >= :createdAfter and group_hearing.created < :createdBefore " +
-        "       and hearing_day.court_code = :courtCode " +
-        "       group by group_hearing.HEARING_ID " +
-        "       ) grouped_hearings " +
-        "on h.id = grouped_hearings.max_id " +
         "inner join hearing_day hday on hday.fk_hearing_id = h.id " +
-        "where hday.hearing_day = :hearingDay " +
-        "and h.deleted = false ",
+        "where h.created >= :createdAfter and h.created < :createdBefore " +
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepository.java:57]: 😌

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/OffenderRepositoryFacade.java [hunk 3]
```diff
@@ -25,21 +25,21 @@ public OffenderEntity save(OffenderEntity offenderUpdate) {
     public OffenderEntity updateOffenderIfItExists(OffenderEntity updatedOffender) {
         return offenderRepository.findByCrn(updatedOffender.getCrn())
             .map(existingOffender -> {
-                // Implementation note: We're breaking immutability here because otherwise Hibernate will
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/OffenderRepositoryFacade.java:28]: 🔥

File: src/test/resources/sql/before-HearingRepositoryFacadeIntTest.sql [hunk 4]
```diff
@@ -1,239 +1,244 @@
...
-
-INSERT INTO courtcaseservicetest.case_comments(id, case_id, comment, "author", created, created_by, created_by_uuid) VALUES (-1700028900, '727af2a3-f9ec-4544-b5ef-2ec3ec0fcf2b', 'PSR in progress', 'Author One', now(), 'before-test.sql', 'fb9a3bbf-360b-48d1-bdd6-b9292f9a0d81');
-INSERT INTO courtcaseservicetest.case_comments(id, case_id, comment, "author", created, deleted, created_by, created_by_uuid) VALUES (-1700028901, '727af2a3-f9ec-4544-b5ef-2ec3ec0fcf2b', 'PSR completed', 'Author One', now(), true, 'before-test.sql', 'fb9a3bbf-360b-48d1-bdd6-b9292f9a0d81');
+INSERT INTO OFFENDER (id, crn, probation_status, previously_known_termination_date, suspended_sentence_order, breach, pre_sentence_activity, awaiting_psr, created_by, version)
+VALUES (-100, 'X25829', 'CURRENT', '2010-01-01', true, true, true, true, 'before-test', 1);
+
+INSERT INTO OFFENDER (id, crn, probation_status, previously_known_termination_date, suspended_sentence_order, breach, pre_sentence_activity, awaiting_psr, created_by)
+VALUES (-99, 'X12345', 'CURRENT', '2010-01-01', true, true, true, true, 'before-test');
+
+-- -- Initial creation of Ferris Bueller
```
Comment (jhevans) [inline src/test/resources/sql/before-HearingRepositoryFacadeIntTest.sql:7]: Can this be deleted?

File: src/test/resources/sql/before-HearingRepositoryFacadeIntTest.sql [hunk 5]
```diff
@@ -1,239 +1,244 @@
...
-VALUES (-196, -196, 'B10JQ', 1, '2022-03-23', '09:00', '2022-03-23 16:59:59.000');
-INSERT INTO courtcaseservicetest.DEFENDANT (id, DEFENDANT_ID, PERSON_ID, defendant_name, name, address, type, date_of_birth, crn, sex, created)
-VALUES (-196, '62d57ee8-a7a6-4b36-857d-8ced9e2aac9b', '0759b0a6-4b9f-11ed-bdc3-0242ac120002', 'Mr Hearing Moved In', '{"title": "Mr", "surname": "BUELLER", "forename1": "Ferris"}', '{"line1": "27", "line2": "Elm Place", "postcode": "ad21 5dr", "line3": "Bangor", "line4": null, "line5": null}', 'PERSON', '1958-10-10', null, 'MALE', '2022-03-23 16:59:59.000');
-INSERT INTO courtcaseservicetest.HEARING_DEFENDANT (id, fk_hearing_id, created, defendant_id)
-VALUES (-196, -196, '2022-03-23 16:59:59.000', '62d57ee8-a7a6-4b36-857d-8ced9e2aac9b');
-INSERT INTO courtcaseservicetest.OFFENCE (ID, FK_HEARING_DEFENDANT_ID, TITLE, SUMMARY, ACT, SEQUENCE, CREATED)
-VALUES (-196, -196, 'Theft from a shop', 'On 01/01/2015 at own, stole article, to the value of £987.00, belonging to person.', 'Contrary to section 1(1) and 7 of the Theft Act 1968.', 1, '2022-03-23 16:59:59.000');
+-- INSERT INTO court_case (id, case_id, case_no, created, source_type)
```
Comment (jhevans) [inline src/test/resources/sql/before-HearingRepositoryFacadeIntTest.sql:71]: Deletable?

File: src/test/resources/sql/before-HearingRepositoryFacadeIntTest.sql [hunk 6]
```diff
@@ -1,239 +1,244 @@
...
-INSERT INTO courtcaseservicetest.DEFENDANT (id, DEFENDANT_ID, PERSON_ID,defendant_name, name, address, type, date_of_birth, crn, sex, created)
-VALUES (-194, '58e61be2-d535-4d8c-a121-30e104fffb7d', '0047279c-4ba1-11ed-bdc3-0242ac120002', 'Mr Hearing Moved Out', '{"title": "Mr", "surname": "BUELLER", "forename1": "Ferris"}', '{"line1": "27", "line2": "Elm Place", "postcode": "ad21 5dr", "line3": "Bangor", "line4": null, "line5": null}', 'PERSON', '1958-10-10', null, 'MALE', '2022-03-23 16:59:59.000');
-INSERT INTO courtcaseservicetest.HEARING_DEFENDANT (id, fk_hearing_id, created, defendant_id)
-VALUES (-194, -194, '2022-03-23 16:59:59.000', '58e61be2-d535-4d8c-a121-30e104fffb7d');
-INSERT INTO courtcaseservicetest.OFFENCE (ID, FK_HEARING_DEFENDANT_ID, TITLE, SUMMARY, ACT, SEQUENCE, CREATED)
-VALUES (-194, -194, 'Theft from a shop', 'On 01/01/2015 at own, stole article, to the value of £987.00, belonging to person.', 'Contrary to section 1(1) and 7 of the Theft Act 1968.', 1, '2022-03-23 16:59:59.000');
--- Update
-INSERT INTO courtcaseservicetest.court_case (id, case_id, case_no, created, source_type)
+-- INSERT INTO court_case (id, case_id, case_no, created, source_type)
```
Comment (jhevans) [inline src/test/resources/sql/before-HearingRepositoryFacadeIntTest.sql:100]: Same again - and the others

**Review by jhevans — DISMISSED** [review]

File: src/test/java/uk/gov/justice/probation/courtcaseservice/service/ImmutableCourtCaseServiceTest.java [hunk 7]
```diff
@@ -113,7 +112,7 @@ private HearingDefendantEntity buildHearingDefendant(String defendantId, Offende
     @ExtendWith(MockitoExtension.class)
     @Nested
     @DisplayName("Tests for createCase by case ID with ExtendedCourtCaseRequest")
-    @Disabled
+    @Disabled("ImmutableCourtCaseService.createHearing is deprecated")
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/service/ImmutableCourtCaseServiceTest.java:115]: Added to the cleanup activities

**Review by jhevans — APPROVED** [review]

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingDefendantEntity.java [hunk 8]
```diff
@@ -56,9 +46,9 @@ public class HearingDefendantEntity extends BaseImmutableEntity implements Seria
     @Column(name = "DEFENDANT_ID", nullable = false)
     private final String defendantId;
```
Comment (vb-moj) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingDefendantEntity.java:47]: Don't think we need this anymore?

Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingDefendantEntity.java:47]: Changes not covered and are not critical are captured here and this is one of them https://dsdmoj.atlassian.net/browse/PIC-2622

**Pull request merged**